### PR TITLE
`ultralytics 8.0.125` NMS speed improvements

### LINKED ISSUE #3450: function export_engine in class Exporter
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Export

### Bug

 def export_engine(self, prefix=colorstr('TensorRT:')):
        """YOLOv8 TensorRT export https://developer.nvidia.com/tensorrt."""
        assert self.im.device.type != 'cpu', "export running on CPU but must be on GPU, i.e. use 'device=0'"
        try:
            import tensorrt as trt  # noqa
        except ImportError:
            if LINUX:
                check_requirements('nvidia-tensorrt', cmds='-U --index-url https://pypi.ngc.nvidia.com')
            import tensorrt as trt  # noqa

        check_version(trt.__version__, '7.0.0', hard=True)  # require tensorrt>=8.0.0
        self.args.simplify = True
        f_onnx, _ = self.export_onnx()

in this function check_version '7.0.0' -> '8.0.0' 

### Environment

_No response_

### Minimal Reproducible Example

_No response_

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by glenn-jocher**
### Summary
📝🐛🚀

This pull request improves NMS speed by up to 1/3. It also updates the documentation of the segmentation feature and fixes a typo in the export feature of the YOLOv5 framework. It adds a tip box and a clarification to the `docs/datasets/segment/index.md` file and corrects the version check for TensorRT in the `ultralytics/yolo/engine/exporter.py` file.

> _`YOLOv5` docs change_
> _Tip box and clarification_
> _Help users in fall_

### Walkthrough
* Fix typo in version check for TensorRT in `export_engine` function ([link](https://github.com/ultralytics/ultralytics/pull/3463/files?diff=unified&w=0#diff-89bbf0a641e1fc4302b73181ac7fccc773b5213720cae9b5cae8aa2393c982fdL450-R450))
* Add tip box and clarification to documentation of YOLO format for segmentation labels in `docs/datasets/segment/index.md` ([link](https://github.com/ultralytics/ultralytics/pull/3463/files?diff=unified&w=0#diff-6b13bc55bc5a9d5cdf6ba001dc56deca2a5455bef9abdc6cadd766058ea02868L38-R42))

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhancements to documentation, library version bump, image processing, and non_max_suppression efficiency in Ultralytics.

### 📊 Key Changes
- **Documentation Update**: Clarified requirements and tips for creating segmentation datasets.
- **Examples Section Adjusted**: Modified external link text to maintain uniform naming conventions.
- **Version Increment**: Updated Ultralytics library version to `8.0.125`.
- **Image Resizing Commented Code**: Comment in `autobackend.py` suggesting use of `Image.BILINEAR` instead of `Image.ANTIALIAS`.
- **TensorRT Version Requirement Correction**: Corrected required TensorRT version check comment to `>=7.0.0`.
- **Code Efficiency**: Improved efficiency in non_max_suppression method by removing an unnecessary cloning operation.

### 🎯 Purpose & Impact
- Ensuring users create segmentation datasets correctly by specifying the minimum number of points for labels.
- The readme update about `.NET ONNX ImageSharp` improves clarity for interested developers.
- The library version increment signals improvements and bug fixes have been made, offering users the latest features.
- Adjusting image resizing may hint at potential future enhancements.
- The TensorRT version requirement correction ensures that users have accurate information regarding dependencies.
- Streamlined `non_max_suppression` function can slightly reduce memory usage and improve runtime performance.

### CODE REVIEW

**Comment by sanmai-NL** [comment]
@glenn-jocher This change seems to have no impact on the source code semantics other than a version bump.

**Comment by AyushExel** [comment]
This looks good. Maybe what Sanmai meant was that the title of the PR should have been something like "bump version" rather than what has changed. That can be written in the release

**Comment by sanmai-NL** [comment]
What's the relevance of the other changes? This PR isn't a clean version bump and I'm looking for the actual changes that improve NMS speed.

File: ultralytics/yolo/engine/exporter.py [hunk 1]
```diff
@@ -447,7 +447,7 @@ def export_engine(self, prefix=colorstr('TensorRT:')):
                 check_requirements('nvidia-tensorrt', cmds='-U --index-url https://pypi.ngc.nvidia.com')
             import tensorrt as trt  # noqa
 
-        check_version(trt.__version__, '7.0.0', hard=True)  # require tensorrt>=8.0.0
+        check_version(trt.__version__, '7.0.0', hard=True)  # require tensorrt>=7.0.0
```
Comment (sanmai-NL) [inline ultralytics/yolo/engine/exporter.py:450]: @AyushExel Is the lack of actual version bump here correct?

Comment (AyushExel) [reply ultralytics/yolo/engine/exporter.py:450]: Yup. It should be similar to https://github.com/ultralytics/ultralytics/pull/3423/files

**Comment by glenn-jocher** [comment]
@sanmai-NL thank you for your question! 

In the PR you mentioned, along with the version bump, there are actually improvements to the NMS (Non-Maximum Suppression) speed in YOLOv8. These improvements result in faster processing and better overall performance when performing NMS, which is an essential step in object detection.

Additionally, the PR includes some documentation updates. These updates clarify the segmentation feature and fix a typo in the export feature of YOLOv8.

I hope this clarifies the relevance of the other changes in the PR. If you have further questions or need more specific details, please let me know.

**Pull request merged**